linux/Platform.c

```diff
@@ -681,6 +695,115 @@ bool Platform_getNetworkIO(NetworkIOData* data) {
    return true;
 }
 
+void Platform_getBacktrace(Vector* frames, const BacktracePanel* panel, char** error) {
+#ifdef HAVE_LIBUNWIND_PTRACE
+   *error = NULL;
+
+   unw_addr_space_t addrSpace = unw_create_addr_space(&_UPT_accessors, 0);
+   if (!addrSpace) {
+      xAsprintf(error, "Unable to initialize libunwind.");
+      return;
+   }
+
+   const pid_t pid = Process_getPid(panel->process);
+
+   if (pid <= 0) {
+      xAsprintf(error, "Unable to get the pid");
+      goto addr_space_error;
+   }
+
+   if (ptrace(PTRACE_ATTACH, pid, 0, 0) == -1) {
+      int ptraceErrno = errno;
+      xAsprintf(error, "ptrace: %s (%d)", strerror(ptraceErrno), ptraceErrno);
+      goto addr_space_error;
+   }
+
+   int waitStatus = 0;
+   if (wait(&waitStatus) == -1) {
+      int waitErrno = errno;
+      xAsprintf(error, "wait: %s (%d)", strerror(waitErrno), waitErrno);
+      goto ptrace_error;
+   }
+
+   if (WIFSTOPPED(waitStatus) == 0) {
+      *error = xStrdup("The process chosen is not stopped correctly.");
+      goto ptrace_error;
+   }
+
+   struct UPT_info* context = _UPT_create(pid);
+   if (!context) {
+      xAsprintf(error, "Unable to create the context of libunwind-ptrace");
+      goto ptrace_error;
+   }
+
+   unw_cursor_t cursor;
+   int ret = unw_init_remote(&cursor, addrSpace, context);
+   if (ret < 0) {
+      xAsprintf(error, "libunwind cursor: ret=%d", ret);
+      goto context_error;
+   }
+
+   int index = 0;
+   do {
+      char procName[2048] = "?";
+      unw_word_t offset;
+      unw_word_t pc;
+
+      BacktraceFrame* frame = BacktraceFrame_new(panel);
+      frame->index = index;
+      if (unw_get_proc_name(&cursor, procName, sizeof(procName), &offset) == 0) {
+         ret = unw_get_reg(&cursor, UNW_REG_IP, &pc);
+         if (ret < 0) {
+            xAsprintf(error, "unable to get program counter register: %d", ret);
+            BacktraceFrame_delete((Object *)frame);
+            break;
+         }
+
+         frame->address = pc;
+         frame->offset = offset;
+         frame->isSignalFrame = unw_is_signal_frame(&cursor);
+
+         frame->functionName = xStrndup(procName, 2048);
+
+#if defined(HAVE_LIBIBERTY)
+         char* demangledName = cplus_demangle(frame->functionName,
+                                              DMGL_PARAMS | AUTO_DEMANGLING);
+         frame->demangleFunctionName = demangledName;
+#endif
+
+#if defined(HAVE_LIBUNWIND_ELF_FILENAME)
+         unw_word_t offsetElfFileName;
+         char elfFileName[2048] = { 0 };
+         if (unw_get_elf_filename(&cursor, elfFileName, sizeof(elfFileName), &offsetElfFileName) == 0) {
+            frame->objectPath = xStrndup(elfFileName, 2048);
+
+            char *lastSplash = strrchr(frame->objectPath, '/');
```

Comment:
I thought you mean "slash".
```suggestion
            char* lastSlash = strrchr(frame->objectPath, '/');
```

Reply:
Yes :smile: . Thank you!